src/NuGet.Core/NuGet.Configuration/Settings/SettingFactory.cs

```diff
@@ -176,7 +176,7 @@ public int GetHashCode(SettingElement obj)
             if (xElement.Name.LocalName.Equals(ConfigurationConstants.PackageSourceMapping, StringComparison.OrdinalIgnoreCase) && duplicatedDescendants != null)
             {
                 var duplicatedKey = string.Join(", ", duplicatedDescendants.Select(d => d.Attributes["key"]));
-                var source = duplicatedDescendants.Select(d => d.Origin.ConfigFilePath).First();
+                var source = duplicatedDescendants.Select(d => d.Origin?.ConfigFilePath).Where(d => d is not null).First();
```

Comment:
In theory `d.Origin` will never be null, because that means that the instance wasn't loaded from a file. But we don't have APIs to load settings from an in-memory stream, and other parts of the code throw NullReferenceException if settings classes are created programatically, rather than from loading from a file.

But technically this could be a change in behaviour, whereas the goal of this PR was no behaviour changes, only null annotations.